chia/cmds/cmds_util.py

```diff
@@ -207,18 +206,18 @@ async def get_wallet(root_path: Path, wallet_client: WalletRpcClient, fingerprin
     return selected_fingerprint
 
 
-async def execute_with_wallet(
+@asynccontextmanager
+async def get_wallet_client(
     wallet_rpc_port: Optional[int],
-    fingerprint: int,
-    extra_params: Dict[str, Any],
-    function: Callable[[Dict[str, Any], WalletRpcClient, int], Awaitable[None]],
-) -> None:
-    async with get_any_service_client(WalletRpcClient, wallet_rpc_port) as (wallet_client, _):
+    fingerprint: Optional[int] = None,
+    root_path: Path = DEFAULT_ROOT_PATH,
+) -> AsyncIterator[Tuple[Optional[WalletRpcClient], int, Dict[str, Any]]]:
+    async with get_any_service_client(WalletRpcClient, wallet_rpc_port, root_path) as (wallet_client, config):
         if wallet_client is None:
-            return
-
-        new_fp = await get_wallet(DEFAULT_ROOT_PATH, wallet_client, fingerprint)
-        if new_fp is None:
-            return
-
-        await function(extra_params, wallet_client, new_fp)
+            yield None, 0, config
+        else:
+            new_fp = await get_wallet(root_path, wallet_client, fingerprint)
+            if new_fp is None:
+                yield None, 0, config
+            else:
+                yield wallet_client, new_fp, config
```

Comment:
this change (`execute_with_wallet` -> `get_wallet_client()` context manager) seems to be independent from the main change in this PR. It's also not obvious that it's an improvement. You gain an extra level of indentation in the commands for instance.
It seems reasonable to make this a separate PR (or drop it if it's not important).

Or is the issue that the previous function would have to pass through variadic argument list? That's simple in python though I would think. Perhaps mypy doesn't support figuring out the types when passing through a generic function like this

Reply:
yeah mypy does not support figuring out the types, so that's why we do it this way,

Reply:
I could remove this change but it would make every other change more lines and just a mess.

Reply:
right. if mypy can't see through it anyway, it kind of defeats the purpose of the patch